app.js

```diff
@@ -31,6 +33,22 @@ client.on('messageDelete', async msg => {
   msg.channel.send(`${author} deleted message: ${message}`);
 });
 
+client.on('presenceUpdate', (oldPresence, newPresence) => {
+  if(oldPresence === undefined) {
+    return;
+  }
+  if(oldPresence.status === newPresence.status) {
+    return;
+  } 
+  if(oldPresence.status !== 'offline' && oldPresence.status !== 'idle') {
+    return;
+  }
+  if(newPresence.status !== 'online') {
+    return;
+  }
```

Comment:
It might be cleaner to consolidate some of this logic into just:
```
if(oldPresence === undefined || oldPresence.equals(newPresence)) { return }

and  

if(oldPresence.status === 'dnd' || newPresence.status !== 'online') { return; }
```

Reply:
> It might be cleaner to consolidate some of this logic into just:
> 
> ```
> if(oldPresence === undefined || oldPresence.equals(newPresence)) { return }
> 
> and  
> 
> if(oldPresence.status === 'dnd' || newPresence.status !== 'online') { return; }
> ```

If I change that logic like you said I think there'll be a bunch of gap cases for custom statuses or everytime people switch off a game it'll say they're online again also you can't compare the presence objects as a whole because something about their presence besides their status could have changed so it won't get caught by oldPresence.equals(newPresence)), I only want it to send the message after coming from offline or idle

Reply:
So I think custom statuses are considered `activities` in the Presence object, `.status` can only be one of these four: https://discord.js.org/#/docs/discord.js/stable/typedef/PresenceStatus.

And the .equals on the Presence object basically just does `oldPresence.status === newPresence.status` just with additional checks

 https://github.com/discordjs/discord.js/blob/stable/src/structures/Presence.js#L122

Reply:
Ok first part makes sense

Reply:
Second part though those additonal checks mess it up no?
Consider the case
 I'm online playing COD
 I stop playing cod
 My presence updates my oldPresence and newPresence both have a status of online but they have different activities lengths
 oldPresence === undefined is false oldPresence.equals(newPresence)) is false
 oldPresence.status === 'dnd' is false newPresence !== 'online' is false
 So it sends an online message when it shouldn't
 
 this.activities.length === presence.activities.length https://github.com/discordjs/discord.js/blob/stable/src/structures/Presence.js#L122
 this line here would make the comparison false

Reply:
oohhh i see what you mean. Yeah that makes sense `oldPresence.status === newPresence.status` is the right way 👍